lib/geoengineer/resources/aws/iam/aws_iam_role_policy_attachment.rb

```diff
@@ -30,18 +30,31 @@ def to_terraform_state
     tfstate
   end
 
+  def fetch_entities(policy_arn, marker = nil, roles = [])
+    params = { policy_arn: policy_arn }
+    params[:marker] = marker unless marker.nil?
+
+    response = AwsClients.iam(provider).list_entities_for_policy(params)
+
+    if response.is_truncated
+      fetch_entities(policy_arn, response.marker, roles + response.policy_roles)
+    else
+      roles + response.policy_roles
+    end
+  end
+
   def remote_resource_params
     return {} unless _policy
     return {} unless _policy.remote_resource
 
     arn = _policy.remote_resource._terraform_id
-    entities = AwsClients.iam(provider).list_entities_for_policy({ policy_arn: arn })
+    attached_roles = fetch_entities(arn)
```

Comment:
because this call is made for every resource this will become super expensive quickly. Can we cache the results on a class hash variable using the policy arn as the key.